setup.cfg

```diff
@@ -0,0 +1,52 @@
+[metadata]
+name = broadbean
+version = 0.10.0
+maintainer = QCoDeS Core Developers
+maintainer_email = qcodes-support@microsoft.com
+description = Package for easily generating and manipulating signal 
+                pulses. Developed for use with qubits in the quantum 
+                computing labs of Copenhagen, Delft, and Sydney, but 
+                should be generally useable.
+keywords = Pulsebuilding signal processing arbitrary waveforms
+long_description = file: README.md
+long_description_content_type = text/markdown
+url = https://github.com/QCoDeS/broadbean
+classifiers =
+    Development Status :: 4 - Beta
+    Intended Audience :: Science/Research
+    License :: OSI Approved :: MIT License
+    Programming Language :: Python :: 3 :: Only
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+    Topic :: Scientific/Engineering
+license = MIT
+
+[options.package_data]
+broadbean =
+    py.typed
+
+[options]
+include_package_data = True
+packages=find:
+python_requires = >=3.7
+# We might as well require what we know will work
+# although older numpy and matplotlib version will probably work too
+install_requires = 
+    numpy>=1.12.1
+    matplotlib
+    schema
+    versioningit>=1.1.1
+
+[options.extras_require]
+test =
+    pytest>=6.2.2
+    pytest-cov>=3.0.0
+    coverage[toml]>=6.2
+    black>=22.3.0
+    mypy>=0.960
+    types-pytz>=2021.3.0
+    jupyter>=1.0.0
+    hypothesis>=5.49.*,<7.0.0
```

Comment:
we should probably kjeep the test_requirement.txt up to date with this? i.e. pin dependencies in test_requirement and keep it more relaxed here? or pin test dependecies here and remove test_requirement alltogether? if we keep the test_requirement file, then we are dependabot-ready, so perhaps that would be my preference :)

Reply:
It seems strange that hypothsis is listed in requirements.txt

Reply:
@trevormorgan could you we also add a comment why we restrict hypothesis to <7?

Reply:
>@trevormorgan could you we also add a comment why we restrict hypothesis to <7?

Coped from another project but it should be needed, I removed that restriction